### ISSUE TITLE
[v2.51] Fail to amend a commit

### ISSUE THREAD

**Issue by pmiossec** [gitextensions/gitextensions#4417]
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**

When trying to amend a commit, I got the error:

```
"C:\Program Files\Git\bin\git.exe" commit --amend -F "C:\Users\phil\Documents\Data\Dev\github\gitextensions\.git\COMMITMESSAGE"
fatal: Unable to create 'C:/Users/phil/Documents/Data/Dev/github/gitextensions/.git/index.lock': File exists.

Another git process seems to be running in this repository, e.g.
an editor opened by 'git commit'. Please make sure all processes
are terminated then try again. If it still fails, a git process
may have crashed in this repository earlier:
remove the file manually to continue.
```

But:
* there is not git process running
* there is no '.git/index.lock' existing

Am I the only one?

**Environment you encounter the issue:**

- GitExtensions version: 2.51
- GIT version:  2.15.1.windows.2
- OS version: Win10

**Did this work in previous version of GitExtensions (which)?**
All previous...

**Comment by pmiossec** [comment]
That does it only with this repository :(

**Comment by gerhardol** [comment]
Duplicate of #3823 
Workaround: Deactivate commit count
This should work better now when #4327 is merged

**Comment by pmiossec** [comment]
@gerhardol Disableing the option didn't solve the problem.
Even with #4327, I have the problem :(

But the command line is working...

**Comment by gerhardol** [comment]
Check command log. What is running?

**Comment by pmiossec** [comment]
```
2018-02-03T22:46:52.1205738+01:00    "C:\Program Files\Git\bin\git.exe" commit --amend -F "C:\Users\phil\Documents\Data\Dev\github\gitextensions\.git\COMMITMESSAGE"    #(took 167 ms)
2018-02-03T22:46:51.7706420+01:00    "C:\Program Files\Git\bin\git.exe" status --porcelain -z --untracked-files    #(took 582 ms)
```

It seems that the 2nd command start just before but end just after...

**Comment by pmiossec** [comment]
It's OK, I found from where the problem come from....

I have enabled some days or weeks ago the feature "Refresh dialog on form focus".
When you try to amend a commit, you've got a confirmation message that steal the focus to the commit form.
So, when the popup is closed and the commit form is refreshed, launching git command in async.
The git command launch enter in conflict with the git commit command.

I have disabled the option now that I know ;)
I will close the issue and open one just to keep a trace...
